util/resolver/resolver.go

```diff
@@ -38,6 +45,31 @@ func NewResolveOptionsFunc(m map[string]RegistryConf) ResolveOptionsFunc {
 			}
 		}
 
+		if c.CA != "" {
+			// if a CA is specified for this registry, add it to the trusted CA pool.
+			rootCAs, _ := x509.SystemCertPool()
+			if rootCAs == nil {
+				rootCAs = x509.NewCertPool()
+			}
+			certs, err := ioutil.ReadFile(c.CA)
+			if err != nil {
+				log.Fatalf("Failed to append %q to RootCAs: %v", c.CA, err)
```

Comment:
Could maybe check this after configuration is loaded as well to report an error to the user on invalid config. Later we can change this function to return an error.